components/econet/econet.cpp

```diff
@@ -65,10 +65,25 @@ void extract_obj_names(const uint8_t *pdata, uint8_t data_len, std::vector<std::
       obj_names->push_back(s);
       break;
     }
-    // Add all bytes until the first occurrence of 0x00
-    std::string s((const char *) start, end - start);
-    obj_names->push_back(s);
-    start = end + 1;
+    // If the value after the end is not a 0 ... the string actually continues...)
+    if (*(end + 1) != 0) {
+      const uint8_t *start2 = end + 1;
+      // Look for the next occurrence of 0x00 in the remaining bytes
+      const uint8_t *end2 = (const uint8_t *) memchr(end + 1, 0, num);
+      // Add all bytes until the first occurrence of 0x00
+      std::string s1((const char *) start, end - start);
+      // Add all bytes until the next occurrence of 0x00
+      std::string s2((const char *) start2, end2 - start2);
+      // Put the strings together and replace the offending 0x00 with a "~"
```

Comment:
I don't feel replacing 0x00 in the middle of strings with ~ is necessary so I didn't do that in the simplified version. Let them get printed shorter. If you disagree feel free to do that in a follow up PR

Reply:
Yep, you got it. Just chose "~" to distinguish it from hopefully anything else that is actually used.

Removing it would make it look like "PRESLIQ"
Replacing it with "_" would make it look like "PRES_LIQ"
Replacing it with "0" (0x30) would make it look like "PRES0LIQ"
I would say we should replace it with something that will tell people something unique / off is happening here. (which led me to "~", but there's likely a better valid ASCII but clearly invalid character.

Reply:
I'm good with the approach of just starting with it being stripped out.